File: mw-kreator-regul-biznesowych/app-modul-model/src/main/java/model/encje/ParametrRegulyEncja.java
package model.encje;

import org.jetbrains.annotations.NotNull;

import javax.persistence.Entity;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import java.util.List;

@Entity
@Table(name="parametry_reguly")
public class ParametrRegulyEncja extends Encja {

    private String nazwa;
    private String typ;
    private String wartoscDomyslna;

    @ManyToOne
    private RegulaEncja regula;


    public ParametrRegulyEncja(String nazwa) {
        this.nazwa = nazwa;
    }

    public ParametrRegulyEncja() {
    }

    public ParametrRegulyEncja(String nazwa, String typ, String wartoscDomyslna) {
        this.nazwa = nazwa;
        this.typ = typ;
        this.wartoscDomyslna = wartoscDomyslna;
    }


    public String getNazwa() {
        return nazwa;
    }

    public void setNazwa(String nazwa) {
        this.nazwa = nazwa;
    }

    public String getTyp() {
        return typ;
    }

    public void setTyp(String typ) {
        this.typ = typ;
    }

    public String getWartoscDomyslna() {
        return wartoscDomyslna;
    }

    public void setWartoscDomyslna(String wartoscDomyslna) {
        this.wartoscDomyslna = wartoscDomyslna;
    }

    public RegulaEncja getRegula() {
        return regula;
    }

    public void setRegula(RegulaEncja regula) {
        this.regula = regula;
    }
}

File: mw-kreator-regul-biznesowych/app-modul-model/src/main/kotlin/model/encje/ParametrRegulyEncja.kt
package model.encje

import javax.persistence.*

@Entity
@Table(name="parametry_reguly")
open class ParametrRegulyEncja(
        val nazwa: String,
        var typ:String?=null,
        val wartoscDomyslna:String?=null
) :Encja(),IWalidowalny {


    @ManyToOne
    lateinit var regula: RegulaEncja

    constructor(aNazwa: String, regula: RegulaEncja) : this(nazwa=aNazwa) {
        this.regula=regula
    }

    @Transient
    override fun waliduj(): List<String> {
        val bledy:MutableList<String> = mutableListOf<String>()
        if(typ==null){
            bledy.add(zglosBladBrakuOkreslonegoTypu(nazwa))
        }
        return bledy
    }
}